src/coord/src/coord.rs

```diff
@@ -917,7 +914,13 @@ impl Coordinator {
                 // connector, which means there is external state (like
                 // a Kafka topic) that's been created on our behalf. If
                 // we fail now, we'll leak that external state.
-                if self.catalog.try_get_by_id(id).is_some() {
+                if let Some(entry) = self.catalog.try_get_by_id(id) {
+                    let sink_instance = entry.sink().expect("connecting to sink").compute_instance;
+                    assert_eq!(
+                        sink_instance, compute_instance,
+                        "connecting to sink on unexpected instance"
+                    );
```

Comment:
I think here you can just plumb through the `compute_instance` from L908?

Reply:
I'm maybe being paranoid in ensuring we're doing the right thing, but will remove these asserts